modules/core/src/v2/coins/eth.ts

```diff
@@ -1196,6 +1208,119 @@ export class Eth extends BaseCoin {
+    ) {
+      throw new InvalidEthereumBatchData('batch data must be a hex string');
+    }
+    prebuildRecipients[0].data = optionalDeps.ethUtil.addHexPrefix(batchData).toLowerCase();
+
+    if (typeof sendAmount !== 'string' || !sendAmount.match(/^[0-9]+$/)) {
+      throw new InvalidAmountError('total amount to send not in valid string form');
+    }
+
+    if (sendAddress !== common.Environments[this.bitgo.env].batcherContractAddress) {
+      throw new InvalidEthereumBatchData('batch transactions must be addressed to the Batcher');
+    }
+
+    const functionSignatureHash = batchData.slice(2, 10);
+    if (functionSignatureHash !== Eth.batcherFunctionSignature) {
+      throw new InvalidEthereumBatchData('batch data function signature hash does not match expected');
+    }
+
+    const [addresses, amounts] = optionalDeps.ethAbi.rawDecode(
+      Eth.batchMethodTypes,
+      Buffer.from(batchData.slice(10), 'hex')
+    );
+    if (addresses.length !== amounts.length) {
+      throw new InvalidEthereumBatchData('recipients and amounts arrays must be the same size');
+    }
+
+    const batchRecipients = [];
+    let totalAmount = new BigNumber(0);
+    for (let i = 0; i < addresses.length; i++) {
+      const address = addresses[i].toString('hex');
+      if (!this.isValidAddress(address)) {
+        throw new InvalidAddressError('invalid address: ' + address);
+      }
+
+      const amount = amounts[i].toString();
+      if (typeof amount !== 'string' || !amount.match(/^[0-9]+$/)) {
+        throw new InvalidAmountError('invalid amount: ' + amounts[i].toString());
+      }
+      totalAmount = totalAmount.plus(amount);
+
+      batchRecipients.push({
+        address,
+        amount,
+      });
+    }
+
+    const sortedBatchRecipients = _(batchRecipients)
+      .sortBy(['address', 'amount'])
+      .map(({ address, amount }) => ({
+        address: optionalDeps.ethUtil.addHexPrefix(address),
+        amount: new BigNumber(amount).toFixed(),
+      }))
+      .value();
+
+    const sortedOriginalRecipients = _(originalRecipients)
+      .sortBy(['address', 'amount'])
+      .map(({ address, amount }) => ({
+        address: optionalDeps.ethUtil.addHexPrefix(address),
+        amount: new BigNumber(amount).toFixed(),
+      }))
+      .value();
+
+    if (!_.isEqual(sortedBatchRecipients, sortedOriginalRecipients)) {
+      throw new InvalidEthereumBatchData('batch recipients in data does not match original recipients');
+    }
+
+    if (!totalAmount.eq(sendAmount)) {
+      throw new InvalidEthereumBatchData(
+        `Batcher send amount ${sendAmount} is not equal to sum of individual transfers ${totalAmount.toFixed()}`
+      );
+    }
+  }
+
+  /**
+   * Validates that the prebuild data returned by platform is valid
+   * @param params
+   * @param callback
+   * @throws Error if the prebuild is invalid
+   */
+  verifyTransaction(params: any, callback?: NodeCallback<any>): Bluebird<any> {
+    return co(function*() {
```

Comment:
add `const self = this;` above coroutine.

Reply:
Fixed.